### ISSUE TITLE
Resizing FloatPanel does not trigger resizing of child panels

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#6157]
I'm on the current `main` branch of Panel.

In https://discourse.holoviz.org/t/auto-resizing-a-holoviews-plot-within-a-floatpanel/6678 banesullivan-kb reports that children of a FloatPanel do not resize when the FloatPanel resizes.

```python
from bokeh.plotting import figure
import panel as pn
pn.extension('floatpanel')

p1 = figure(height=250, sizing_mode='stretch_width', margin=5)
p2 = figure(height=250, sizing_mode='stretch_width', margin=5)

p1.line([1, 2, 3], [1, 2, 3])
p2.circle([1, 2, 3], [1, 2, 3])

f = pn.layout.FloatPanel(p1, pn.layout.Divider(), p2, name="Responsive", sizing_mode='stretch_width')
pn.Column('**Example: Basic `FloatPanel`**', f, height=850, width=1000).servable()
```

I can reproduce this issue in the video below. You will notice that the Bokeh plots resize when I trigger "Inspect".

https://github.com/holoviz/panel/assets/42288570/2779f9f0-20c9-4dc9-a402-cd98f939eb4d

**Comment by banesullivan-kobold** [comment]
Thank you so much for promptly addressing this! I haven't had a chance to test out the fix, but it looks like it should do the trick for this issue.

I want to follow up with a few other fixes we ended up implementing on our end. 

Full credit to @jerry-kobold on this one.

The following fixed a few additional things for use, including:

- programmatically position the float panel
- set the z-index when a float panel is clicked or interacted with to make sure it comes to the front
- dynamically update the title

@jerry-kobold may have more to say on the implementation here:

```py
class EnhancedFloatPanel(FloatPanel):
    title = param.String(default=None)

    current_w = param.Integer(default=None, bounds=(None, None))
    current_h = param.Integer(default=None, bounds=(None, None))
    position_x = param.Integer(default=None, bounds=(None, None))
    position_y = param.Integer(default=None, bounds=(None, None))
    z_index = param.Integer(default=100, bounds=(0, None))

    reset = param.Event()
    front = param.Event()

    _scripts = {
        **FloatPanel._scripts,
        **{
            "render": """
        if (state.panel) {
          view.run_script('close')
        }
        var config = {
          headerTitle: data.name,
          content: float,
          theme: data.theme,
          id: 'jsPanel'+data.id,
          position: view.run_script('get_position'),
          contentSize: `${model.width} ${model.height}`,
          onstatuschange: function() {
            data.status = this.status
          },
          onbeforeclose: function() {
           data.status = 'closed'
           return true
          },
          // a hook to update the z-index of the panel. for some reason, the multiple panels do not seem to know about
          // each other so they all think they're the top panel and thus don't properly move themselves to the front.
          // we do a round-trip into python land where we do know about all the panels so we can manually update their
          // z-index by triggering the `front` event, which then updates the z-index
          onfronted: function(panel, status) {
            data.z_index = parseInt(panel.style.zIndex)
            data.front = true
          },
          // this hook updates some python-side parameters to track the dimensions of the panel after resize. this is
          // needed so that we can update the corresponding plot with the appropriate dimensions. not using the original
          // height and width params here because who knows, they might be used for something else
          resizeit: {
            resize: function(panel, paneldata, event) {
              data.current_w = Math.ceil(paneldata.width)
              data.current_h = Math.ceil(paneldata.height)
            }
          },
          // the below hook is necessary because otherwise the offset parameters don't get updated when the panel is
          // dragged around. if these parameters are not updated, then events which expect their values to be different
          // will not fire, as from the python side these will remain unchanged despite the fact that the panel has
          // been moved. to fix this we simply update the offset params with the final location of the panel after it
          // has been dragged.
          dragit: {
            stop: function(panel, paneldata, event) {
              data.position_x = Math.ceil(paneldata.left)
              data.position_y = Math.ceil(paneldata.top)
            }
          }

        }
        if (data.contained) {
          config.container = view.container
        }
        config = {...config, ...data.config}
        state.panel = jsPanel.create(config);
        if (data.status !== 'normalized') {
           view.run_script('status')
        }
        """,
            "z_index": """
            // we are using the hook that is connected to a changing z-index on the python
            // model to update the z-index of the panel style. this is kind of gross but
            // for reasons that i do not fully understand (i blame the shadow dom), the
            // different panels are not aware of each other and thus front() has no effect
            // because each panel already thinks itself to be the front. so we do a round
            // trip into python and update this manually.
            state.panel.style.zIndex = data.z_index
            """,
            "title": """
            // using the "name" attribute doesn't work for this, so I made the "title" attribute instead
            state.panel.setHeaderTitle(data.title)
            """,
            "reset": """
            if (data.reset == true) {
                view.run_script('reposition');
                data.reset = false
            }
            """,
        },
    }
```

It'd be awesome if these additions could be included directly in panel and we're probably game to contribute a small PR for this if welcome

**Comment by jerry-kobold** [comment]
Thanks @banesullivan-kobold. As you can see from this code and the comments, I implemented a couple of gross workarounds to accomplish what I was trying to do and make no claims to the overall quality of this solution. I can confirm that it does work, but, with the z-index, you have to know which panels you are tracking on the Python side. In our case, each panel contains a bokeh plot and they can be added or removed dynamically so we have to keep track of them. For the rest, I'm mostly just leveraging the various already-existing hooks in `jsPanel`.

**Comment by lachlanjs** [comment]
Hi there,
I just added a reply to the [discourse thread from which this originated](https://discourse.holoviz.org/t/auto-resizing-a-holoviews-plot-within-a-floatpanel/6678/5), but reiterating here.

If you change "stretch_width" to "stretch_both", then the plots become squished vertically. They will respond to changes in width, but not in height. There is still a bug here.

It's tricky to me that it does work for the width but not for the height...

**Comment by lachlanjs** [comment]
@MarcSkovMadsen @philippjfr 
I don't have necessary permissions to reopen this myself but it was suggested to me on the discourse thread that it should be reopened. I don't see much worth in creating a new issue considering the only change to the minimal reproducible example is changing "stretch_width" to "stretch_both".

For now I believe the following is a bandaid fix:
- Use some of the code provided by Kobold et. al here above, specifically the `current_w` and `current_h` bit.
- Take the components that you want to put inside your FloatPanel, and put them inside a Column inside your FloatPanel.
- Link the `current_w` and `current_h` of the enhanced float panel to the `width` and `height` parameters of your Column. I think you need to use `sizing_mode="fixed"` also for the Column.

I will try this soon and get back, but there will still be the underlying bug

**Comment by lachlanjs** [comment]
The method I suggested works. It may be necessary to bound the width of the float column inside the float panel to be greater than zero to avoid errors. I also chose to subtract ~60 from the height of the float panel (I think because the float panel height includes the draggable bar at the top containing the maximisation etc. controls)

I am now running into separate issues with the float panel (updating Holoviews plots on zoom/pan within the float panel) which I will try to resolve. I will create a separate issue for this if necessary.
